R/pkg/R/sparkR.R

```diff
@@ -365,6 +365,23 @@ sparkR.session <- function(
     }
     overrideEnvs(sparkConfigMap, paramMap)
   }
+  if (!grepl(".*shell\\.R$", Sys.getenv("R_PROFILE_USER"), perl = TRUE)) {
```

Comment:
There's a programmatic check in Spark JVM side you might want to do that instead of hard coding a file name.